rs/sns/governance/src/governance.rs

```diff
@@ -4626,7 +4760,29 @@ impl Governance {
         self.process_proposals();
 
         if self.should_check_upgrade_status() {
-            self.check_upgrade_status().await;
+            self.upgrade_periodic_task_lock
+                .try_run_isolated_upgrade_task(
+                    UpgradeTaskInstance::check_upgrade_status(),
+                    || async {
+                        // Governance::check_upgrade_status()
+                        println!("DO NOT MERGE; self.proto = {:?}", self.proto);
+                    },
+                    ic_cdk::api::time,
+                )
+                .await;
+        }
+
+        if self.should_refresh_cached_upgrade_steps() {
+            self.upgrade_periodic_task_lock
+                .try_run_isolated_upgrade_task(
+                    UpgradeTaskInstance::refresh_cached_upgrade_steps(),
+                    || async {
+                        // Governance::refresh_cached_upgrade_steps,
+                        println!("DO NOT MERGE; self.proto = {:?}", self.proto);
+                    },
+                    ic_cdk::api::time,
```

Comment:
This is problematic for now since it can't be run in tests :(

Reply:
Here's what Swap is using for reading the current time: 

```
fn now_nanoseconds() -> u64 {
    if cfg!(target_arch = "wasm32") {
        time()
    } else {
        SystemTime::now()
            .duration_since(SystemTime::UNIX_EPOCH)
            .expect("Failed to get time since epoch")
            .as_nanos()
            .try_into()
            .expect("Failed to convert time to u64")
    }
}
```

Could we use something similar in this canister?